### ISSUE TITLE
Remap command line <C-p> and <C-n> to act like <Up> and <Down>

### ISSUE THREAD

**Issue by craigmac** [neovim/neovim#16637]
### Feature already in Vim?

No

### Feature description

`<Up>` and `<Down>` when in command line will respect what you've already written on the command line so far, and find the previous/next match from history. I think this is common enough remapping (learned this one from a Drew Neil book), that is more useful improvement over default behaviour. 

It would be a remap like this:

```
cnoremap <C-p> <Up>
cnoremap <C-n> <Down>
```

I'm not sure of anything depending on default `<C-p>/<C-n>` behaviour, which is to simply cycle through the command line history, without using what's already on the line for context.

**Comment by zeertzjq** [comment]
Simply mapping to `<Up>` and `<Down>` will disable another functionality of `<C-P>`/`<C-N>`: iterate through command line completion. This should be better:
```vim
cnoremap <expr> <C-P> wildmenumode() ? "<C-P>" : "<Up>"
cnoremap <expr> <C-N> wildmenumode() ? "<C-N>" : "<Down>"
```

**Comment by justinmk** [comment]
> `<Up>` and `<Down>` when in command line will respect what you've already written on the command line so far,

`<M-p>` / `<M-n>` already do this by default.

**Comment by craigmac** [comment]
@justinmk I've just tried Terminal.app, Alacritty, Kitty, and Wezterm with `nvim --clean`, `<M-p>` doesn't do this, at least on macOS (maybe a system thing?).  In Alacritty, with and without tmux running, I do:

```
nvim --clean
:edit /tmp/test
:echo 'hi'
:echo 'hello'
:ed<Up> => :edit /tmp/test
:ed<C-p> => :ed (nothing given)
:ed<M-p> => E353: Nothing in register "
```

iTerm2 with same steps: same result (E353)
wezterm: same result
kitty: same same
.. you get the idea 

`:h c_META` explains why, `<M-p>` is sending `<Esc>p` (exiting cmdline and trying to paste from unnamed register).
Neovim could map <M-p> and <M-n> to do what I suggested above, instead of having to press <Up>/<Down>, but that isn't the default behaviour of Neovim at the moment.
